velox/exec/MergeJoin.cpp

```diff
@@ -118,7 +118,7 @@ void MergeJoin::initializeFilter(
 BlockingReason MergeJoin::isBlocked(ContinueFuture* future) {
   if (future_.valid()) {
     *future = std::move(future_);
-    return BlockingReason::kWaitForExchange;
+    return BlockingReason::kWaitForProducer;
```

Comment:
Using this reason for MergeJoin might be confusing. Technically, MergeJoin has 2 inputs / producers, but it gets blocked only when there is not data on the "right" side. It might be better to use a separate reason for MergeJoin.

Reply:
@mbasmanova 
Good point, I saw that and was thinking "why not kWaitForJoinBuild?".
But since I prefer executing such changes with minimum splash damage area I kept it consistent.
So, in your opinion, what reason should we use here?
kWaitForJoinBuild or something else?

Reply:
I think we need a new block reason. Maybe, kWaitForMergeJoinRightSide